### PR TITLE
feat(utils, module-sdk, medusa): Extract pg connection utils to utils package

### PR DESCRIPTION
**Pull request by adrien2p**
Move knex connection initialisation to the utils package and update the places that are using it

### CODE REVIEW

File: packages/medusa/src/loaders/pg-connection.ts [hunk 1]
```diff
@@ -1,53 +1,39 @@
...
-      database,
-      ssl: extra?.ssl ?? false,
-      idle_in_transaction_session_timeout:
-        extra.idle_in_transaction_session_timeout ?? undefined, // prevent null to be passed
-    }
-
-    // TODO: see later if it is possible to use the same connection with multiple pool config using this shared connection across the modules
-    // const pgConnection = await new Client(config).connect() // or any other method to create your connection here
+  delete extra.max
+  delete extra.idleTimeoutMillis
```
Comment (olivermrbl) [inline packages/medusa/src/loaders/pg-connection.ts:23]: thought: This seems to be a leaking abstraction. Should this not be handled in the mikro util?

File: packages/medusa/src/loaders/pg-connection.ts [hunk 2]
```diff
@@ -1,53 +1,39 @@
...
-        idleTimeoutMillis: extra.idleTimeoutMillis ?? undefined, // prevent null to be passed
-      },
-    })
+  const pgConnection = ModulesSdkUtils.createPgConnection({
+    clientUrl: connectionString,
+    schema,
+    driverOptions: extra,
+    pool: {
+      max: poolMax,
+      idleTimeoutMillis,
```
Comment (olivermrbl) [inline packages/medusa/src/loaders/pg-connection.ts:31]: same as other comment

File: packages/types/src/modules-sdk/index.ts [hunk 3]
```diff
@@ -201,6 +201,18 @@ export interface ModuleServiceInitializeOptions {
     database?: string
     driverOptions?: Record<string, unknown>
     debug?: boolean
+    pool?: {
```
Comment (olivermrbl) [inline packages/types/src/modules-sdk/index.ts]: same as other comment

File: packages/medusa/src/loaders/pg-connection.ts (hunk 1 above)
Comment (adrien2p) [reply packages/medusa/src/loaders/pg-connection.ts:23]: the extra comes from the module config from medusa-config, that is why I re manipulate it here to pass it down to the funtion with the expected shape

File: packages/medusa/src/loaders/pg-connection.ts (hunk 2 above)
Comment (adrien2p) [reply packages/medusa/src/loaders/pg-connection.ts:31]: the db connection and the pool are two different things, that is why they are separated. what do you suggest?

File: packages/medusa/src/loaders/pg-connection.ts (hunk 1 above)
Comment (olivermrbl) [reply packages/medusa/src/loaders/pg-connection.ts:23]: What I mean is, should we change the `database_extra` to just be a generic object, pass it to the `ModulesSdkUtil.createPgConnection` and do the data shaping close to where it is used.

Does this make sense? 😅

Comment (olivermrbl) [reply packages/medusa/src/loaders/pg-connection.ts:23]: Actually nevermind. This is nitpicking, at this point :)

File: packages/types/src/modules-sdk/index.ts (hunk 3 above)
Comment (olivermrbl) [reply packages/types/src/modules-sdk/index.ts]: Something tells me it is wrong to add all these knex-specific options to the module config type, as developers would be able to use other connections i.e. other types of pools no?

File: packages/medusa/src/loaders/pg-connection.ts (hunk 1 above)
Comment (adrien2p) [reply packages/medusa/src/loaders/pg-connection.ts:23]: I see what you mean, but i think the util define what it accept as the input and then the consumer prepare its object shape according to the expectation of the util and not the other way around. Or maybe i ve not properly understood what you mean 😅

File: packages/types/src/modules-sdk/index.ts (hunk 3 above)
Comment (adrien2p) [reply packages/types/src/modules-sdk/index.ts]: Yes for sure, maybe we should just have a Record type and reference the knex api doc for us?

Comment (olivermrbl) [reply packages/types/src/modules-sdk/index.ts]: Yeah, that could make sense

**Review by olivermrbl — APPROVED** [review]
LGTM!

**Pull request merged**